react/components/__tests__/QuantitySelector.test.tsx

```diff
@@ -0,0 +1,240 @@
+import React from 'react'
+import { render, screen, fireEvent } from '@vtex/test-tools/react'
+import { ToastContext } from '@vtex/styleguide/lib/ToastProvider'
+
+import QuantitySelector from '../QuantitySelector'
+
+describe('<QuantitySelector />', () => {
+  it('should change from dropdown to input after quantity exceeds max value', () => {
+    const onChange = jest.fn()
+
+    render(
+      <QuantitySelector id="1" value={5} maxValue={50} onChange={onChange} />
+    )
+
+    const [dropdown] = screen.getAllByRole('combobox')
+
+    expect(dropdown).toBeInTheDocument()
+
+    expect(dropdown).toHaveValue('5')
+
+    fireEvent.change(dropdown, { target: { value: '10' } })
+
+    expect(onChange).toHaveBeenLastCalledWith(10)
+    expect(dropdown).not.toBeInTheDocument()
+
+    const [input] = screen.getAllByRole('textbox')
+
+    expect(input).toHaveValue('10')
+  })
+
+  it('should change back to input after changing the value to under max value', () => {
```

Comment:
Not sure if I understood that well, but isn't the point of this test to go from input -> combobox?
```suggestion
  it('should change back to dropdown after changing the value to under max value', () => {
```

Reply:
nice